app/components/UI/Input.js

```diff
@@ -140,12 +151,20 @@ class Input extends React.Component {
             )}
           </Label>
         )}
+
         <SystemInput
+          opacity={readOnly ? 0.6 : null}
```

Comment:
Probably would be a good idea to extract this into a styled-comp to keep it cleaner

Reply:
I've moved the `outline` and `opacity ` style props over to the styled component.

Unfortunately, I can't move the dynamic border color props to the same place as the utility methods that calculate the correct color  need access to the `formState` prop, which doesn't get passed on to the styled component. So I've had to leave those applied via the `css` prop.

Reply:
Maybe I'm missing something but why can't you path form props down to styled component and map it with the same function you already have?

Reply:
The styled component in this case is a rebas system component, wrapped in a styled component (see https://github.com/LN-Zap/zap-desktop/pull/1378/files#diff-9406bc9468bcaded7a7aec88441c833eR49), which outputs an `input` dom element.

For example, here is a stripped down version of what we have (psudocode)

First we create a system component, and wrap it with a rebass system component.
```js
const StyledInput = styled(
  system(
    {
      as: 'input',
      border: 1,
      borderColor: 'gray',
      borderRadius: 5,
    },
    'borders',
    'borderColor',
    'borderRadius',
  )
)`
  opacity: ${props => (props.disabled || props.readOnly ? '0.6' : 'inherit')};
  outline: none;
`
```

Then we use it in our Input component:

```js
// Integrate the field with informed.
const Input = asField(({
   fieldApi,
   fieldState,
   ...rest
 }) => {
return {
  <StyledInput
    css={Object.assign(
      {
        'border-color': mapDefaultBorderColor(this.props),
        '&:not([readOnly]):not([disabled]):focus': {
          'border-color': mapFocusBorderColor(this.props)
        }
      },
      css
    )}
    {...rest}
    />
})
```

`mapDefaultBorderColor` needs access to things that are not being passed on to the `StyledInput`, such as `formState`.

If I pass additional props such as `formState` to the styled component, that are not intended to end up on the underlaying dom element (`input` in this case) I will get warnings about unknown properties being set in the dom.

I'm not sure how I could pass the props to the styled component purely for the purpose of calculating the css value, and then strip those properties out so that they don't get passed onto the dom. I could possibly do it by introducing another intermediary component but it seems like overkill.

Am I missing something?

Reply:
@korhaliv I've updated this as per our slack chat. Works like a charm :)